app/src/main/java/com/cp/tasky/auth/shared/data/repository/RemoteDataSource.kt

```diff
@@ -0,0 +1,21 @@
+package com.cp.tasky.auth.shared.data.repository
+
+import com.cp.tasky.auth.shared.data.remote.AuthApi
+import com.cp.tasky.auth.login.data.remote.models.LoginApiBody
+import com.cp.tasky.auth.login.data.remote.models.LoginApiResponse
+import javax.inject.Inject
+
+interface RemoteDataSource {
+
+    suspend fun loginUser(loginApiBody: LoginApiBody): LoginApiResponse
+}
```

Comment:
Is there any value you gain from having this abstraction? In my experience, you rarely need abstractions where the abstraction and implementation are in the same layer. An abstraction only makes sense if you have or plan to have at least two implementations of it. Which other implementation do you plan to have here?

Reply:
I am planning to have registerUser here. Also, I want to have the abstraction to ease the testing process.